core/primitives-core/src/code.rs

```diff
@@ -28,4 +30,15 @@ impl ContractCode {
     pub fn clone_for_tests(&self) -> Self {
         Self { code: self.code.clone(), hash: self.hash }
     }
+
+    /// Destructs this instance and returns the code.
+    pub fn take_code(self) -> Vec<u8> {
+        self.code
+    }
+}
+
+impl Debug for ContractCode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ContractCode").field("hash", &self.hash).finish()
```

Comment:
nit: would be nice to include code size as part of debug info

Reply:
done